lib/jobs/ipmi-job.js

```diff
@@ -199,6 +224,98 @@ function ipmiJobFactory(
     };
 
     /**
+     * Collect SEL entries list from IPMI and add two properties 
+     * ("Sensor Type Code" and "Event Type Code") to each sel entry
+     * @memberOf IpmiJob
+     *
+     * @param data
+     */
+    IpmiJob.prototype.collectIpmiSelEntries = function(data) {
+        var self = this;
+        var stringifiedEntries = "";
+        var lastUpdatedSelDate;
+        var currentSelDate;
+        var lastReportedSelEntryID;
+        var totalEntries;
+        var sel;
+
+        return Promise.all([
+           self.collectIpmiSelInformation(data),
+           waterline.workitems.findOne({ id: data.workItemId })
+        ])
+        .spread (function(selInfo, workObj){
+            totalEntries = parseInt(selInfo.Entries);
+            var tmpDate = selInfo["Last Add Time"].split(/\/|:| /);
+            tmpDate = tmpDate[0] +" "+ tmpDate[1] +", "+ tmpDate[2] +
+                " "+ tmpDate[3] +":"+ tmpDate[4]+":"+ tmpDate[5];
+            currentSelDate = new Date(tmpDate);
+            lastReportedSelEntryID = _.get(workObj, 
+                'config.lastReportedSelEntryID', 0);
+            lastUpdatedSelDate = _.get(workObj, 
+                'config.lastReportedSelEntryID', new Date("01 01, 0100 00:00:00"));
+
+            // reset the counter when the the SEL log has been cleared
+            if(totalEntries < lastReportedSelEntryID){
+                lastReportedSelEntryID = 0;
+            }
+            stringifiedEntries = _.range(lastReportedSelEntryID + 1, totalEntries + 1).join(' ');
+            if(currentSelDate > lastUpdatedSelDate) {
+               return self.getSelEntries(data, stringifiedEntries)
+               .tap(function() {
+                   workObj.config.lastReportedSelEntryID = totalEntries;
+                   workObj.config.lastUpdatedSelDate = currentSelDate;
+                   return workObj.save();
+               });
+            }
+            return [];
+        })
+        .then(function(unparsedSel) {
+            return parser.parseSelDataEntries(unparsedSel);
+        })
+        .then(function(parsedSel) {
+            var rawArr = [];
+            var rawData;
+            // we get the raw sel info for each sel entry in order to decode additional informaiton
+            sel = parsedSel;
+            if(sel) {
+                _.forEach(parsedSel, function(entry) {
+                    rawData = self.getRawSelEntry(data, entry["SEL Record ID"])
+                    .then(function(data){
+                        return data;
+                    });
+                    rawArr.push(rawData);
+                });
+            }
+            return Promise.all(rawArr);
```

Comment:
@jlongever @yyscamper  Couldn't we use `Promise.map` to avoid this problem?
```
rawArr = Promise.map(parsedSel, function(entry) {
    return self.getRawSelEntry(data, entry["SEL Record ID"]);
});
```